lib/src/components/table.dart

```diff
@@ -214,8 +214,8 @@ class _ShadTableState extends State<ShadTable> {
 
   int get effectiveRowCount =>
       widget.rowCount +
-      (widget.header != null ? 1 : 0) +
-      (widget.footer != null ? 1 : 0);
+      (widget.headerBuilder != null ? 1 : 0) +
+      (widget.footerBuilder != null ? 1 : 0);
```

Comment:
Thanks for the PR @Kyziq 🙏
Here the code was partially correct, you must check both `widget.header` and `widget.headerBuilder`

Reply:
This is needed to have the same behavior when using `ShadTable.list` constructor

Reply:
Thanks for pointing that out! I've updated the code to check both `widget.header` and `widget.headerBuilder` for consistent behavior across both constructors, including `ShadTable.list`. This ensures correct handling of headers and footers